### PR TITLE
feat(client): added interface generation

### PR DESCRIPTION
**Pull request by alesbrelih**
This way its easier to use mock autogenerators to mock clients for tests.

### CODE REVIEW

**Comment by alesbrelih** [comment]
I need to stop polishing template once I generate... Sorry. I fixed it now.

File: gen/_template/client.tmpl [hunk 1]
```diff
@@ -4,6 +4,20 @@
 
 {{- if $.PathsClientEnabled }}
 {{- with $ops := $.Operations }}
+
```
Comment (tdakkota) [inline gen/_template/client.tmpl:7]: Could you split template change and autogenerated code into two commits, as [contributing guide suggests](https://github.com/ogen-go/ogen/blob/main/CONTRIBUTING.md#committing-generated-code)?

File: ogen
Comment (tdakkota) [inline ogen:1]: It seems you've commited ogen binary by accident.

File: gen/_template/client.tmpl [hunk 2]
```diff
@@ -4,6 +4,20 @@
...
+// ClientInterface calls operations described by OpenAPI v3 specification.
+type ClientInterface interface {
```
Comment (tdakkota) [inline gen/_template/client.tmpl:9]: nitpick: I feel like
```suggestion
type Invoker interface {
```
sounds better

**Review by tdakkota — CHANGES_REQUESTED** [review]

File: gen/_template/client.tmpl (hunk 1 above)
Comment (alesbrelih) [reply gen/_template/client.tmpl:7]: Sure 👍 Didn't see that document, my bad. Now they are split.

File: ogen
Comment (alesbrelih) [reply ogen:1]: Wops, I'm testing locally with my own project, my bad. Removed now.

File: gen/_template/client.tmpl (hunk 2 above)
Comment (alesbrelih) [reply gen/_template/client.tmpl]: Way better 👍 Changed and regenerated code.

**Review by tdakkota — APPROVED** [review]

**Pull request merged**